### ISSUE TITLE
TextPanel Block doesnt refresh text to display

### ISSUE THREAD

**Issue by ChrisonSimtian** [KeenSoftwareHouse/SpaceEngineers#431]
TextPanel-Block seems to be broken...
It doesnt display text which was added eventhough it saves it to the Text-String (either public or private)
Workaround: manually refresh the block (e.g. change font size and force the block to redraw the display area)

Todo: Check whether this is only for TextPanel or if this also affects LCD Displays

**Comment by lord-devious** [comment]
Hi there :) Looks like a very old bug might be back. However this is not the bug report section. You need to head back to the forums for that :smile:

**Comment by ChrisonSimtian** [comment]
yeah I'm doing both right now :-) I figured out that this is related to the way I'm calling the PB.
Calling it through the GUI works just fine, calling it via a sensor block e.g. doesnt work unless I use aboves workaround

As a reference:
http://forums.keenswh.com/threads/textpanels-broken.7370066/

**Comment by mexmer** [comment]
after you edit text on panel, you should always call ShowPublicTextOnScreen() or ShowPrivateTextOnScreen() to refresh LCD content on all clients.

editing panel value trough PB doesn't trigger refresh, it will only update value, and set flag, which is processed lately by main thread.

**Comment by ChrisonSimtian** [comment]
At least it wasnt that way for the last couple of patches (I used an existing script in an existing world which stopped displaying text)

Anyways if thats the expected behaviour shouldnt RefreshRenderText() be exposed instead of having to call either of both methods? (depending on which text you currently display...)

**Comment by mexmer** [comment]
idk, i always used showtextonscreen, because that was only way, to get text to display immediately on DS.

also lot of scripts in shop calls showtextonscreen, it's rather rare to see script, that doesn't call it.

**Comment by ChrisonSimtian** [comment]
I see your point :-)

Anyways I'd like to leave this issue open to have a reference for a potential future change in that particular class and handle it in terms of a feature/change request rather than pointing out an actual problem.

I hope thats the proper way to deal with this kind of change management?? If not I'm very comfortable to close this issue and just deliver a PR in the future :-)

**Comment by cheerkin** [comment]
That's very diasppointing, because this hack is quite expensive. This problem has emerged for me several updates ago, and I've found that using this method consumes around 10-15ms. So my assembler management script runs for 52ms per tick instead of 2ms (I have to refresh multiple panels per tick). Combined stuttering from a couple of scripts makes gaming experience quite unpleasant.
